pkg/agent/openflow/pipeline.go

```diff
@@ -1821,22 +1822,6 @@ func (c *client) snatCommonFlows(nodeIP net.IP, localSubnet net.IPNet, localGate
 			Action().GotoTable(nextTable).
 			Cookie(c.cookieAllocator.Request(category).Raw()).
 			Done(),
-		// This flow is for the return traffic of connections to a local
-		// Pod through the gateway interface (so gatewayCTMark is set).
-		// For example, the return traffic of a connection from an IP
-		// address (not the Node's management IP or gateway interface IP
-		// which are covered by other flows already) of the local Node
-		// to a local Pod. It might also catch the Service return
-		// traffic from a local server Pod, but this case is also
-		// covered by other flows (the flows matching the local and
-		// remote Pod subnets) anyway.
-		l3FwdTable.BuildFlow(priorityNormal).
```

Comment:
Why remove this flow?

See:
> 		// This flow is for the return traffic of connections to a local
> 		// Pod through the gateway interface (so gatewayCTMark is set).
> 		// For example, the return traffic of a connection from an IP
> 		// address (not the Node's management IP or gateway interface IP
> 		// which are covered by other flows already) of the local Node
> 		// to a local Pod.

Reply:
I mentioned the reason in the commit message "It also removes a flow in L3Forwarding table that specially handles gatewayCT related traffic, which has been taken care of by another more generic flow in same table." It's added in https://github.com/antrea-io/antrea/pull/2318.

https://github.com/antrea-io/antrea/blob/4405aaab0e075ac571307ef975146e003f74e223/pkg/agent/openflow/pipeline.go#L1184-L1204

I think currently the deleted flow will never be hit:
```
table=70, priority=210,ct_state=+rpl+trk,ct_mark=0x20,ip actions=mod_dl_dst:e2:e5:a4:9b:1c:b1,goto_table:80
table=70, priority=200,ip,reg0=0x2/0xffff,ct_mark=0x20 actions=goto_table:80
```

Reply:
Ok, I did not see that! But would you update the comments for the new flow to include the removed case too?

Reply:
Probably better to keep the comments in this func, and saying this case is covered by another flow?

Reply:
Do you think it makes sense to only send non reply traffic to snatTable so we don't need to have special handling for various return traffic? I see the deleted flow is for gatewayCT related return traffic, and the comment also said Service return traffic may be caught by it. I think generally no reply traffic would need to go to snatTable?

Reply:
Yes, but how we identify non-reply traffic?

Reply:
I think "-rpl+trk" should work?

Reply:
That sounds good to me, but I did not think through all cases to be confident.

Reply:
When I was trying to ensure there would be no regression if only forwarding "-rpl+trk" only traffic to snatTable, I didn't figure out whether it would really affect windows case. As the patch needs to be backported, to minimize the risk, I would just make minimal change to the case of local pod subnet to fix the specific issue. Maybe let's consider the windows case together later.
I have added a comment here to say the case of gatewayCT flow has been covered by other flow, please take another look.

Reply:
The strategy works for me.